.github/workflows/maven.yml

```diff
@@ -30,6 +32,20 @@ jobs:
           distribution: 'temurin'
           cache: maven
       - name: Build with Maven
-        run: ./mvnw $mvn_options verify -PnoGpg --file pom.xml
+        run: ./mvnw $mvn_options install -PnoGpg --file pom.xml
```

Comment:
we need `install` since during tests with downstream project these artifacts will be reused

Reply:
What would you think of turning all the tests of Datafaker into a test artifact, and run them as part of the downstream project?

Reply:
I guess most of them are independent whether they are running within Datafaker code or some downstream projects.
There are only few of them where it works differently.

Also there will be another challenge: how to make it possible
I see several solutions and none of them is perfect
1. copy-paste of tests however it means we need to maintain 2 repos synced at least from tests point of view
2. we could publish `test-jar` and reuse it in downstream project however anyway need to create a bunch of child classes and keep it sync (e.g. each time a new test class is appearing in Datafaker we need to create a corresponding child in third party)
3. something else that I don't know yet

Reply:
Sorry, I don't understand point 2. Why would we need to create child classes? The idea I had was to indeed create a test artifact, put all the tests in there, and use surefire to run the tests against the jar. I'm not sure if we do some magic which prevents this?

Alternatively, I think we need only 1 test, which is very similar to the repeatability test : just call every in the public api and make sure it returns something. That would have stopped a few bugs from being released.

Reply:
May be I missed something however I thought that tests will not run if we just add a `test-jar` dependency without any usage of that

Reply:
Well, here it says we could run them if we package them up and give them to surefire, unless I'm misunderstanding something:

https://stackoverflow.com/questions/10496846/run-junit-tests-contained-in-dependency-jar-using-maven-surefire

Reply:
may be makes sense to try it once more 
not sure when will have time for this